perfkitbenchmarker/linux_packages/cassandra.py

```diff
@@ -90,6 +94,27 @@ def AptInstall(vm):
   _Install(vm)
 
 
+def JujuInstall(vm):
+  if vm.isController is True:
+    if vm.vm_group_specs and 'workers' in vm.vm_group_specs:
+      vm.JujuDeploy('cs:trusty/cassandra',
+                    vm.vm_group_specs['workers'].vm_count,
+                    'workers')
```

Comment:
I don't think we should tie the package installation to the benchmark - there could be future benchmarks that will require a cassandra cluster that don't name that group "workers" (right now there are two cassandra benchmarks, both of which happen to have named those groups workers, but this won't be true for all packages/benchmarks).

Instead I'd suggest you slightly rewrite JuJuMixin.Install() and this function:

```
  def Install(self, package_name):
     """Installs a PerfKit package on the VM."""
     package = linux_packages.PACKAGES[package_name]
     try:
       with self.controller.installation_lock: # Make sure another unit doesn't try to install the charm at the same time
         if package_name not in self.controller._installed_packages:
           package.JujuInstall(self.controller, self.group_name)  
           self.controller._installed_packages.add(package_name)
   
       except AttributeError:
         if package_name not in self._installed_packages:
           package.AptInstall(self)
           self._installed_packages.add(package_name)
```
The above assumes you also modify ConstructVirtualMachines to add "group_name" to each vm.
Let me know if you need more clarification.